### PR TITLE
fix: JSONForm fix events/actions controls JS mode produces additional binding strings

### LINKED ISSUE #12543: [Bug]-[4]:JSON Form (Select field) `onFilterUpdate` property shows two different variations of same code block
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Screenshot 2022-04-04 at 12 53 44 PM
Screenshot 2022-04-04 at 12 53 36 PM

### Steps To Reproduce

1. Create JSON form add select field
2. Populate `onFilterUpdate` check and uncheck `JS` toggle.

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by ashit-rath**
## Description

Action/Events control present in JSONForm field configuration, when JS mode is used to write code and toggled it adds additional bindings.
To fix, the `globalContext` for action execution is used to pass `formData`, `sourceData`, `fieldState` and the `JSON_FORM_COMPUTE_VALUE` control is replaced with the default JS mode input

Fixes #12543

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Jest

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/12543-json-form-actions-jsmode 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 56.5 **(-0.02)** | 37.98 **(-0.02)** | 35.85 **(-0.01)** | 56.74 **(-0.01)**
 :green_circle: | app/client/src/components/propertyControls/JSONFormComputeControl.tsx | 25.26 **(0.79)** | 9.76 **(0)** | 11.76 **(0)** | 25 **(0.9)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**
 :red_circle: | app/client/src/widgets/JSONFormWidget/widget/index.tsx | 26.89 **(-1.81)** | 0 **(0)** | 20 **(-0.69)** | 27.19 **(-1.94)**
 :red_circle: | app/client/src/widgets/JSONFormWidget/widget/propertyConfig/helper.ts | 89.06 **(-9.12)** | 73.08 **(-6.09)** | 85.71 **(-6.6)** | 87.93 **(-10.07)**</details>

### CODE REVIEW

File: app/client/src/widgets/JSONFormWidget/widget/index.tsx [hunk 1]
```diff
@@ -261,8 +264,26 @@ class JSONFormWidget extends BaseWidget<
     });
   };
 
+  applyGlobalContextToAction = (actionPayload: ExecuteTriggerPayload) => {
+    const payload = klona(actionPayload);
+    const { globalContext } = payload;
+
+    payload.globalContext = {
+      ...globalContext,
+      ...{
```
Comment (sbalaji1192) [inline app/client/src/widgets/JSONFormWidget/widget/index.tsx:273]: we can simply define formData, fieldState & sourceData in the globalcontext
```
payload.globalContext = {
...globalContext,
formData: this.props.formData,
fieldState: this.props.fieldState,
sourceData: this.props.sourceData
}

**Review by sbalaji1192 — APPROVED** [review]

File: app/client/src/widgets/JSONFormWidget/widget/index.tsx (hunk 1 above)
Comment (ashit-rath) [reply app/client/src/widgets/JSONFormWidget/widget/index.tsx:273]: Ahh yes 😅

**Comment by chandannkumar** [comment]
Tested this PR and working as expected

**Review by sbalaji1192 — APPROVED** [review]

**Pull request merged**